### PR TITLE
feat(dashboard,types,js-sdk): Cleanup collection domain

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds missing collection HttpTypes
- Adds missing sdk functions
- Adds usage of sdk to collection domain.

### CODE REVIEW

File: packages/admin-next/dashboard/src/hooks/api/collections.tsx [hunk 1]
```diff
@@ -90,19 +90,24 @@ export const useUpdateCollection = (
 export const useUpdateCollectionProducts = (
   id: string,
   options?: UseMutationOptions<
-    ProductCollectionRes,
-    Error,
-    UpdateProductCollectionProductsReq
+    { collection: HttpTypes.AdminCollection },
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/hooks/api/collections.tsx:85]: note: I ve seen that we have some types that include `Response` at the end of the types and other are not, we will have to align. not for this pr but more of a general comment :D

File: packages/admin-next/dashboard/src/hooks/api/collections.tsx [hunk 2]
```diff
@@ -90,19 +90,24 @@ export const useUpdateCollection = (
...
+    FetchError,
+    AdminUpdateCollectionProducts
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/hooks/api/collections.tsx]: suggestion: should we take all the types from the HttpTypes to stay coherent through the usage?

**Review by sradevski — APPROVED** [review]
Nice 👍 Approving, but check Adrien's comments

**Review by adrien2p — APPROVED** [review]

File: packages/admin-next/dashboard/src/hooks/api/collections.tsx (hunk 1 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/src/hooks/api/collections.tsx:85]: If we exported the the type as `{ collection: AdminCollection }` then I think it should be suffixed with Response. But none of the other recent ones are exported like that, only the `Admin<Entity>` is exported, so I stuck with that for this PR as well. But yea we need to align on this, for sure.

**Pull request merged**